pkg/receive/handler.go

```diff
@@ -142,6 +148,38 @@ func NewHandler(logger log.Logger, o *Options) *Handler {
 				Help: "The number of times to replicate incoming write requests.",
 			},
 		),
+		writeTimeseriesTotal: promauto.With(o.Registry).NewCounterVec(
+			prometheus.CounterOpts{
+				Namespace: "thanos",
+				Subsystem: "receive",
+				Name:      "write_timeseries_total",
+				Help:      "The number of timeseries received in the incoming write requests.",
+			}, []string{"code", "method", "tenant"},
+		),
+		writeSamplesTotal: promauto.With(o.Registry).NewCounterVec(
+			prometheus.CounterOpts{
+				Namespace: "thanos",
+				Subsystem: "receive",
+				Name:      "write_samples_total",
+				Help:      "The number of sampled received in the incoming write requests.",
+			}, []string{"code", "method", "tenant"},
+		),
+		writeBytesTotal: promauto.With(o.Registry).NewSummaryVec(
+			prometheus.SummaryOpts{
+				Namespace: "thanos",
+				Subsystem: "receive",
+				Name:      "write_bytes",
+				Help:      "The number of bytes received in the body of incoming write requests.",
+			}, []string{"code", "method", "tenant"},
+		),
+		writeInflightHTTPRequests: promauto.With(o.Registry).NewGaugeVec(
+			prometheus.GaugeOpts{
+				Namespace: "thanos",
+				Subsystem: "receive",
+				Name:      "write_requests_inflight",
+				Help:      "The number of inflight write HTTP requests being handled at the same time.",
+			}, []string{"tenant"},
+		),
```

Comment:
Interestingly, these two last (total of bytes per request and amount of inflight HTTP requests), are already exported by the `pkg/extprom/http.defaultInstrumentationMiddleware`. Although such middleware is generic and doesn't include the tenant label, which is important for the Thanos Receive context.

I am not sure what to do with the default instrumentation middleware in this case. I think it should be adapted for Thanos Receive, so that we don't duplicate work to keep track of these metrics. I am looking for feedback/help on this regard.